### PR TITLE
Permutation generation refactoring

### PR DESCRIPTION
**Pull request by hmemcpy**
This refactors permutation generation and combines it with folder-based test environment creation.

### CODE REVIEW

File: src/PerformanceTests/Persistence.V5.InMemory.V5/Persistence.V5.InMemory.csproj [hunk 1]
```diff
@@ -7,8 +7,8 @@
     <ProjectGuid>{D8CB2DA9-4AE4-417C-89C3-00555F5E1F47}</ProjectGuid>
     <OutputType>Library</OutputType>
     <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>Persistence.V5.InMemory.V5</RootNamespace>
-    <AssemblyName>Persistence.V5.InMemory.V5</AssemblyName>
+    <RootNamespace>Persistence.V5.InMemory</RootNamespace>
```
Comment (ramonsmits) [inline src/PerformanceTests/Persistence.V5.InMemory.V5/Persistence.V5.InMemory.csproj]: @hmemcpy Forgot to restore the version number here.

Comment (hmemcpy) [inline src/PerformanceTests/Persistence.V5.InMemory.V5/Persistence.V5.InMemory.csproj]: @ramonsmits Ooops, looks like I forgot to restore others as well. Will fix!

Comment (hmemcpy) [inline src/PerformanceTests/Persistence.V5.InMemory.V5/Persistence.V5.InMemory.csproj]: @ramonsmits I restored the versions (with a small change, replaced dot with underscore) for the downstream dependencies. For the built-in ones (like InMemory and MSMQ) I omitted the version. If we ever put them outside of core, they will probably be versioned at 1.0.0, so until then calling it v6 is confusing.

**Comment by hmemcpy** [comment]
Lots of changes here, but it's ready to be merged.
@WilliamBZA @ramonsmits @HEskandari please review.

**Comment by hmemcpy** [comment]
:tada:

**Pull request merged**